src/target/drivers/filesystems/syscalls.c

```diff
@@ -35,20 +35,25 @@ static FATFS fat[2];
 static FATFS fat[1];
 #endif
 
+#ifdef FIL  // If FIL is #defined, it maps to FATFS
+    #define fil fat
+#else
+    static FIL fil[1];
+#endif
 extern u8 _drive_num;
 
 extern void init_err_handler();
 
 int FS_Mount();
 void FS_Unmount();
 
-long _open_r (FATFS *r, const char *file, int flags, int mode);
-int _close_r (FATFS *r);
+long _open_r (FIL *r, const char *file, int flags, int mode);
+int _close_r (FIL *r);
 
 
-int _read_r (FATFS *r, char * ptr, int len);
-int _write_r (FATFS *r, char * ptr, int len);
-int _lseek_r (FATFS *r, int ptr, int dir);
+int _read_r (FIL *r, char * ptr, int len);
```

Comment:
Lint Error:
```
50:  Extra space before ( in function call  
50:  Use int16/int64/etc, rather than the C type long  
51:  Extra space before ( in function call  
54:  Extra space before ( in function call  
```